tablex.lua

```diff
@@ -202,29 +202,28 @@ function tablex.take_random(t, r)
 	return table.remove(t, tablex.random_index(t, r))
 end
 
---return a random index based on weights provided (or nil if it's empty)
--- { 0.3, 1, 6, 0.5 } -> (3rd index most likely)
+--return a random value from table based on weights provided (or nil if it's empty)
+-- w: weights, t: table to pick value from
 -- possible todo:
 --	provide normalisation outside of this function, require normalised weights
---	provide table of values _and_ weights and return the value
-function tablex.weighted_random(t, r)
-	if #t == 0 then
+function tablex.pick_weighted_random(w, t, r)
```

Comment:
i think `t, w, r` is a better argument order; t is always the first argument in all these functions, the weights feel more like a parameter than the "subject"